src/Application/Items/Commands/RepairUserItemCommand.cs

```diff
@@ -44,7 +44,7 @@ public async Task<Result<UserItemViewModel>> Handle(RepairUserItemCommand req, C
                 return new(CommonErrors.UserItemNotFound(req.UserItemId));
             }
 
-            if (userItem.Rank >= 3)
+            if (userItem.Item!.Rank >= 3)
```

Comment:
This endpoint needs a rewrite that you can do in a following PR. It also needs to unset the broken bool if the user has the money.

Reply:
the unset is already done  userItem.IsBroken = false; line 81 . 
Cleanup planned in another pr

Reply:
I see, still you'll have to remove the loomin part of the command